Изучить и заполнить NetworkApi с необходимыми запросами по экрану поиск

JudjinGM opened the pull request:
Создал пулл реквест для ревью по работе с экраном Поиск.

JudjinGM committed:
1. Добавил добавил необходимые зависисмости в Di.
2. Реализовал поиск(viewModel и fragment).
3. Поправил gradle (не хватало зависимоcти для safeArgs)
4. Поправил стринги (не зватало слово Найдено)
5. Добавил имя для плейсхолдера пустого экрана

JudjinGM committed:
Refactoring #1. Добавил переход на экран фильтров.

JudjinGM committed:
Создал 2 новые модели для ui слоя. Создал к ним маппер

JudjinGM committed:
Merge branch 'feature/TECH-112/vacancy_ui_models' into test

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/common/App.kt

JudjinGM committed:
Refactoring. Поправил ошибки.

JudjinGM committed:
Refactoring №1. Поправил ошибку с номером телефона.

JudjinGM committed:
Refactoring №2. Поправил ошибку отоброжением ЗП не указана

anonim-no commented on app/src/main/java/ru/practicum/android/diploma/search/di/searchDataModule.kt:
```diff
@@ -0,0 +1,27 @@
+val searchDataModule = module {
+
+    single<HeadHunterApiService> {
+        Retrofit.Builder().baseUrl("https://api.hh.ru/")
```
https://api.hh.ru нужно вынести в константу

JudjinGM replied:
окей!

manualcheg commented on app/src/main/java/ru/practicum/android/diploma/search/di/searchDomainModule.kt:
```diff
@@ -0,0 +1,18 @@
+
+val searchDomainModule = module {
+
+    singleOf(::NetworkConnectionProviderImpl) { bind<NetworkConnectionProvider>() }
```
Оказалось, что тут можно ещё убрать лямбду: singleOf(::ClassA) bind InterfaceA::class

JudjinGM replied:
понял, поправлю!

BigMag74 commented on app/src/main/java/ru/practicum/android/diploma/search/ui/viewModel/SearchViewModel.kt:
```diff
@@ -1,10 +1,111 @@
+        setState(SearchState.Success.Empty)
+    }
+
+    fun <T> debounce(delayMillis: Long,
```
у нас же этот метод в common/util лежит

JudjinGM replied:
у меня какая-то проблема с этим методом, не подтягивается из файла

JudjinGM replied:
проблему решил

BigMag74 commented:
![image](https://github.com/anonim-no/practicum-android-diploma/assets/116560396/d48e5d68-7fb3-4108-98f0-c807ca3d14cb)
Не получается последнюю вакансию просмотреть, RV спрятался под навигацией

JudjinGM committed:
Refactoring #3.
1. Добавил interceptor для добавления header.
2. Добавил interceptor для логирования. Теперь можно смотреть что уходит и приходит сразу в logcat.
3. Поправил ошибки и недочеты из ревью.
4. Переместил формирование VacancyUi в презентейшн слой.

JudjinGM committed:
Refactorin #4. Поправил ошибку, что не отображался город в заголовке
